Update cut_its actions

jtangrot opened the pull request:
<!-- markdownlint-disable ul-indent -->
This adresses issue #371 

This PR changes --cut_its to take a string as argument instead of a boolean. Accepted values are currently "none" (default behaviour, no cutting), "full" (cut out whole ITS region and use this for taxonomy assignment, previous behaviour if cut_its was set), and "its2" (cut out ITS2 region only and use that for taxonomy assignment).
Also, the option --its_partial <x> is added, to allow ITSx to report also partial ITS regions if they are longer than the given cutoff <x>. 
For reference, all fasta files, a summary file, and a file listing arguments to ITSx are now saved in the results/itsx folder.

## PR checklist

- [X] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
    - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/ampliseq/tree/master/.github/CONTRIBUTING.md)
    - [ ] If necessary, also make a PR on the nf-core/ampliseq _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [X] Make sure your code lints (`nf-core lint`).
- [X] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [X] Usage Documentation in `docs/usage.md` is updated.
- [X] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

d4straub commented on modules/local/dada2_taxonomy.nf:
```diff
@@ -26,6 +26,9 @@ process DADA2_TAXONOMY {
     set.seed(100) # Initialize random number generator for reproducibility
 
     seq <- getSequences(\"$fasta\", collapse = TRUE, silence = FALSE)
+    if ( length(seq) < 1 ) {
+        stop("assignTaxonomy: No sequences found in fasta file $fasta")
+    }
```
Is that a thing? If this could be due to the empty output of itsx, than I'd rather make process itsx fail instead of this one (which is downstream).

jtangrot replied:
For me it has only been a problem when ITSx doesn't find any ITS regions, so I guess it's reasonable to move the error to the itsx module. Will do that.

d4straub reviewed (commented):
Hi, looks good!
- I have one comment about the error message that I would appreciate if you could address it.
- Also, could you please update the CHANGELOG?
- And lastly, there is a current bug that makes Markdown lint fail, I fixed it in file [.github/workflows/linting.yml](https://github.com/nf-core/ampliseq/pull/385/files#diff-717013d02a8afac81852f25ba8ecd3cc0ca53a7f1fe25bedebfd66d6f518d21c) in the other PR that I had no time yet to perfect and merge. Its just ` node-version: '10'` to `node-version: '16'` (which will raise another error, but that one is easier to deal with).

jtangrot commented:
> Hi, looks good!
> 
> * I have one comment about the error message that I would appreciate if you could address it.
> * Also, could you please update the CHANGELOG?
> * And lastly, there is a current bug that makes Markdown lint fail, I fixed it in file [.github/workflows/linting.yml](https://github.com/nf-core/ampliseq/pull/385/files#diff-717013d02a8afac81852f25ba8ecd3cc0ca53a7f1fe25bedebfd66d6f518d21c) in the other PR that I had no time yet to perfect and merge. Its just ` node-version: '10'` to `node-version: '16'` (which will raise another error, but that one is easier to deal with).

Thanks for your comments and suggestions. I think I have fixed it all!

jtangrot commented:
I'm not sure what to do about the linting errors...

d4straub commented:
About the linting errors:
- `nf-core linting / nf-core (pull_request)` fails with `files_unchanged.github/workflows/linting.yml does not match the template`, thats expected, cause I recommended to modify `linting.yml` (so that `nf-core linting / Markdown (pull_request)` is giving an output), so all good here
- `nf-core linting / Markdown (pull_request)` fails due to usage of `_` and `*`, doesnt matter right now, I also fixed that in https://github.com/nf-core/ampliseq/pull/385 (which I still have 0 time for the final touch before merging)

I can merge your PR anyway (even with failed linting), is that fine to you?

jtangrot commented:
> I can merge your PR anyway (even with failed linting), is that fine to you?
Of course, thanks!!
